tests/callbacks/test_inference.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Test inference APIs.
+"""
+import os
+import tempfile
+from functools import partial
+
+import pytest
+import torch
+from torch.utils.data import DataLoader
+
+from composer.callbacks import ExportForInference
+from composer.models import composer_resnet
+from composer.trainer import Trainer
+from tests.common.datasets import RandomImageDataset
+
+
+@pytest.mark.parametrize(
+    'model_cls, sample_input',
+    [
+        (partial(composer_resnet, 'resnet18'), (torch.rand(1, 3, 224, 224), torch.randint(10, (1,)))),
+    ],
+)
+def test_inference_callback_torchscript(model_cls, sample_input):
+    save_format = 'torchscript'
+    model = model_cls()
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        save_path = os.path.join(tempdir, f'model.pt')
+        export_for_inference = ExportForInference(save_format=save_format, out_filename=str(save_path))
+
+        # Construct the trainer and train
+        trainer = Trainer(
+            model=model,
+            callbacks=export_for_inference,
+            train_dataloader=DataLoader(RandomImageDataset(shape=(3, 224, 224))),
+            max_duration='1ba',
+        )
+        trainer.fit()
+
+        model.eval()
+        orig_out = model(sample_input)
+
+        loaded_model = torch.jit.load(save_path)
+        loaded_model.eval()
+        loaded_model_out = loaded_model(sample_input)
+
+        assert torch.allclose(
```

Comment:
could just use `torch.testing.assert_close` here to give an informative output. (https://pytorch.org/docs/stable/testing.html)

Reply:
Changed to assert_close so both the torchscrip and onnx test match.